libs/common/Networking/TcpNetworkHandlerBase.cs

```diff
@@ -144,22 +151,37 @@ void RecvEventArg_Completed(object sender, SocketAsyncEventArgs e)
                     {
                         // No more things to receive
                         Dispose(e);
-                        break;
+                        return;
                     }
-                    OnNetworkReceive(e.BytesTransferred);
-                    e.SetBuffer(networkReceiveBuffer, networkBytesRead, networkReceiveBuffer.Length - networkBytesRead);
-                } while (!e.AcceptSocket.ReceiveAsync(e));
+                    var receiveTask = OnNetworkReceiveAsync(e.BytesTransferred);
+                    if (!receiveTask.IsCompleted)
```

Comment:
`IsCompletedSuccessfully` is better than `!IsCompleted` because it excludes faulted and canceled tasks, allowing the await to properly capture and rethrow those exceptions.